modules/extensions-support/src/main/java/org/mule/module/extension/internal/introspection/DefaultExtensionFactory.java

```diff
@@ -0,0 +1,203 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.module.extension.internal.introspection;
+
+import static org.mule.module.extension.internal.util.MuleExtensionUtils.alphaSortDescribedList;
+import static org.mule.util.Preconditions.checkArgument;
+import org.mule.api.registry.ServiceRegistry;
+import org.mule.common.MuleVersion;
+import org.mule.extension.introspection.Configuration;
+import org.mule.extension.introspection.DescribingContext;
+import org.mule.extension.introspection.Extension;
+import org.mule.extension.introspection.ExtensionFactory;
+import org.mule.extension.introspection.Operation;
+import org.mule.extension.introspection.Parameter;
+import org.mule.extension.introspection.declaration.ConfigurationDeclaration;
+import org.mule.extension.introspection.declaration.Construct;
+import org.mule.extension.introspection.declaration.Declaration;
+import org.mule.extension.introspection.declaration.OperationDeclaration;
+import org.mule.extension.introspection.declaration.ParameterDeclaration;
+import org.mule.extension.introspection.spi.DescriberPostProcessor;
+import org.mule.module.extension.internal.ImmutableDescribingContext;
+
+import com.google.common.collect.ImmutableList;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Default implementation of {@link ExtensionFactory}
+ *
+ * @since 3.7.0
+ */
+public final class DefaultExtensionFactory implements ExtensionFactory
+{
+
+    private final List<DescriberPostProcessor> postProcessors;
+
+    public DefaultExtensionFactory(ServiceRegistry serviceRegistry)
+    {
+        postProcessors = searchPostProcessors(serviceRegistry);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Extension createFrom(Construct construct)
```

Comment:
Not sure I wrote this in the other PR: extension construction should rely on the declaration , not the construct